### PR TITLE
Make ImageEvaluationArtifact instance close image once GCed

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Make ImageEvaluationArtifact instance close image once GCed

## How is this patch tested?

N/A

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/evaluation/artifacts.py [hunk 1]
```diff
@@ -22,6 +22,9 @@ def _load_content_from_file(self, local_artifact_path):
         self._content = open_image(local_artifact_path)
         return self._content
 
+    def __del__(self):
+        self._content.close()
```
Comment (harupy) [inline mlflow/models/evaluation/artifacts.py:26]: Have you manually tested this method works?

**Review by harupy — APPROVED** [review]
Left a comment, otherwise LGTM

File: mlflow/models/evaluation/artifacts.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (WeichenXu123) [reply mlflow/models/evaluation/artifacts.py:26]: A small unit test code:
```
class A1():
    def __init__(self, a):
        self.a = a
    def __del__(self):
        print(f"del A1 {self.a}")

class B1():
    def __init__(self, a):
        self.a1 = A1(a)
    def __del__(self):
        print(f"del B1")

b1 = B1(3)
b2 = B1(4)

del b1
del b2
```

[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/models/evaluation/artifacts.py:26]: I meant `self._content.close`

[the remark this answers is not in the document]
Comment (WeichenXu123) [reply mlflow/models/evaluation/artifacts.py:26]:
```
>>> from PIL.Image import open as open_image
>>> image = open_image("/tmp/a1.jpg")
>>> image.fp
<_io.BufferedReader name='/tmp/a1.jpg'>
>>> image.close()
>>> image.fp
>>> 
```
close method works

[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/models/evaluation/artifacts.py:26]: got it, thanks for the confirmation!

**Pull request merged**